plugins/inputs/libvirtd/libvirtd.go

```diff
@@ -0,0 +1,194 @@
+package libvirtd
+
+import (
+	"fmt"
+	"github.com/beevik/etree"
+	"github.com/digitalocean/go-libvirt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"net"
+	"time"
+)
+
+const (
+	sampleConfig = `
+  lib_virt_sock = "/var/run/libvirt/libvirt-sock"
+  network_devices = []
+`
+)
+
+// MockPlugin struct should be named the same as the Plugin
+type LibVirtD struct {
+	client         *libvirt.Libvirt
+	LibVirtSock    string
+	DiskDevices    []string
+	NetworkDevices []string
+}
+
+// Description will appear directly above the plugin definition in the config file
+func (l *LibVirtD) Description() string {
+	return `This is an example plugin for libvirt`
+}
+
+// SampleConfig will populate the sample configuration portion of the plugin's configuration
+func (l *LibVirtD) SampleConfig() string {
+	return sampleConfig
+}
+
+func init() {
+	virt := &LibVirtD{}
+	inputs.Add("libvirtd", func() telegraf.Input { return virt })
+}
+
+// Find all the disks in the XML data
+func (l *LibVirtD) getDiskDevices(dom libvirt.Domain, acc telegraf.Accumulator) {
```

Comment:
I wonder if instead of reading through the domain definition to pull out the disks we should just iterate through the  volumes.  This is more familiar to me as a libvirt user, since volumes are one of main parts of the virsh command, while disks are somewhat hidden inside the domain.  Maybe we can avoid parsing xml as a bonus.

Reply:
In our use case for example, we use ceph and none of the vol commands list our disks. I could not find another way around getting the data without looping through the XML. I am rather new to libvirt so there may be another way that I have not figured out yet. Side note, it seems like most virsh commands are built into the libvirtd-go lib.